src/org/traccar/protocol/Jt600ProtocolDecoder.java

```diff
@@ -201,24 +201,25 @@ private Position decodeW01(String sentence, Channel channel, SocketAddress remot
 
     private static final Pattern PATTERN_U01 = new PatternBuilder()
             .text("(")
-            .number("(d+),")                     // id
-            .number("Udd,")                      // type
-            .number("d+,").optional()            // alarm
-            .number("(dd)(dd)(dd),")             // date (ddmmyy)
-            .number("(dd)(dd)(dd),")             // time
-            .expression("([TF]),")               // validity
-            .number("(d+.d+),([NS]),")           // latitude
-            .number("(d+.d+),([EW]),")           // longitude
-            .number("(d+.?d*),")                 // speed
-            .number("(d+),")                     // course
-            .number("(d+),")                     // satellites
-            .number("(d+%),")                    // battery
-            .expression("([01]+),")              // status
-            .number("(d+),")                     // cid
-            .number("(d+),")                     // lac
-            .number("(d+),")                     // gsm signal
-            .number("(d+),")                     // odometer
-            .number("(d+)")                      // index
+            .number("(d+),")                                // id
+            .number("(Udd),")                               // type
+            .number("d+,").optional()                       // alarm
+            .number("(dd)(dd)(dd),")                        // date (ddmmyy)
+            .number("(dd)(dd)(dd),")                        // time
+            .expression("([TF]),")                          // validity
+            .number("(d+.d+),([NS]),")                      // latitude
+            .number("(d+.d+),([EW]),")                      // longitude
+            .number("(d+.?d*),")                            // speed
+            .number("(d+),")                                // course
+            .number("(d+),")                                // satellites
+            .number("(d+%),")                               // battery
+            .expression("([01]+),")                         // status
+            .number("(d+),")                                // cid
+            .number("(d+),")                                // lac
+            .number("(d+),")                                // gsm signal
+            .number("(d+),")                                // odometer
+            .number("(d+),")                                // serial number
+            .expression("([0-9A-F][0-9A-F])").optional()    // checksum
```

Comment:
I think we can use `.number("(xx)?")` instead. Easier to read and won't require to change comment indentation.